ext/standard/array.c

```diff
@@ -5894,11 +5903,24 @@ PHPAPI bool php_array_pick_keys(const php_random_algo *algo, php_random_status *
 	zend_bitset_clear(bitset, bitset_len);
 
 	i = num_req;
+	int failures = 0;
 	while (i) {
 		randval = algo->range(status, 0, num_avail - 1);
-		if (!zend_bitset_in(bitset, randval)) {
+		if (EG(exception)) {
+			goto fail;
+		}
+		if (zend_bitset_in(bitset, randval)) {
+			if (++failures > PHP_RANDOM_RANGE_ATTEMPTS) {
```

Comment:
For some reason CI fails here because it can't find `PHP_RANDOM_RANGE_ATTEMPTS`, even though php_random.h seems to be included. Could be a ccache issue too, I ran into similar oddities because of that in the past.

Reply:
It moved to php_random.h in ac3ecd03af009d433d4b75d570b3b0f0a3fc0ff7, which is 8.3+. Locally I tested with master and just picked the changes. I'll fix that.